flocker/docs/test/test_version.py

```diff
@@ -0,0 +1,195 @@
+
+    def test_weekly_release(self):
+        """
+        When the version is from a weekly release, the documentation version
+        is left unchanged.
+        """
+        self.assertEqual(get_doc_version('0.3.2dev1'), '0.3.2dev1')
+
+    def test_pre_release(self):
+        """
+        When the version is from a pre-release, the documentation version
+        is left unchanged.
+        """
+        self.assertEqual(get_doc_version('0.3.2pre1'), '0.3.2pre1')
+
+    def test_development_vesion(self):
+        """
+        When the version is from a development version, the documentation
+        version is left unchanged.
+        """
+        self.assertEqual(get_doc_version('0.3.2-1-gf661a6a'),
+                         '0.3.2-1-gf661a6a')
+
+    def test_dirty(self):
+        """
+        When the version is dirty, the documentation version is left unchanged.
+        """
+        self.assertEqual(get_doc_version('0.3.2-1-gf661a6a-dirty'),
+                         '0.3.2-1-gf661a6a-dirty')
+
+    def test_doc(self):
+        """
+        When the documentation version is from a documentation release, the
+        trailing '+docX' is stripped.
+        """
+        self.assertEqual(get_doc_version('0.3.2+doc11'), '0.3.2')
+
+    def test_doc_dirty(self):
+        """
+        When the version is from a documentation release but is dirty, the
+        documentation version is left unchanged.
+        """
+        self.assertEqual(get_doc_version('0.3.2+doc0-dirty'),
+                         '0.3.2+doc0-dirty')
+
+
+class IsReleaseTests(SynchronousTestCase):
+    """
+    Tests for :function:`is_release`.
+    """
+
+    def test_marketing_release(self):
+        """
+        When the version is from a marketing release, it is a release.
+        """
+        self.assertTrue(is_release('0.3.2'))
+
+    def test_weekly_release(self):
+        """
+        When the version is from a weekly release, it isn't a release.
+        """
+        self.assertFalse(is_release('0.3.2dev1'))
+
+    def test_pre_release(self):
+        """
+        When the version is from a pre-release, it isn't a release.
+        """
+        self.assertFalse(is_release('0.3.2pre1'))
+
+    def test_development_vesion(self):
+        """
+        When the version is from a development version, it isn't a release.
+        """
+        self.assertFalse(is_release('0.3.2-1-gf661a6a'))
+
+    def test_dirty(self):
+        """
+        When the version is dirty, it isn't a release.
+        """
+        self.assertFalse(is_release('0.3.2-1-gf661a6a-dirty'))
+
+    def test_doc(self):
+        """
+        When the documentation version is from a documentation release, it is a
+        release.  """
+        self.assertTrue(is_release('0.3.2+doc11'))
+
+    def test_doc_dirty(self):
+        """
+        When the version is from a documentation release but is dirty, it isn't
+        a release.
+        """
+        self.assertFalse(is_release('0.3.2+doc0-dirty'))
```

Comment:
:question: The incrementing doc number starts at 1 so this (and the other occurrences of doc0) is not a realistic case.